File: python/src/endpoints/db_insert.py
import os
from worqhat import Worqhat


def create_user() -> None:
    """Create a new user record."""
    # Initialize the client with your API key
    client = Worqhat(
        api_key=os.environ.get("WORQHAT_API_KEY"),  # Using environment variables for security
        environment=os.environ.get("WORQHAT_ENVIRONMENT", "production"),  # Defaults to production
    )

    try:
        response = client.db.insert_record(
            table="users",          # The table to insert into
            data={                  # The data to insert
                "name": "John Doe",
                "email": "john@example.com",
                "role": "user",
                "active": True
            }
        )

        # Handle the successful response
        print(f"User created with ID: {response.data.get('documentId')}")
        print(f"Created user: {response.data}")
    except Exception as e:
        # Handle any errors
        print(f"Error creating user: {str(e)}")


def create_product_with_custom_id() -> None:
    """Create a product with custom document ID."""
    import time

    # Initialize the client with your API key
    client = Worqhat(
        api_key=os.environ.get("WORQHAT_API_KEY"),
        environment=os.environ.get("WORQHAT_ENVIRONMENT", "production"),  # Defaults to production
    )

    # Generate a custom ID
    custom_id = f"prod_{int(time.time())}"  # Using timestamp for unique ID

    try:
        response = client.db.insert_record(
            table="products",       # The table to insert into
            data={                  # The data to insert
                "documentId": custom_id,  # Specify your own document ID
                "name": "Premium Widget",
                "price": 99.99,
                "inStock": True,
                "category": "electronics"
            }
        )

        # Handle the successful response
        print(f"Product created with custom ID: {response.data.get('documentId')}")
        print(f"Created product: {response.data}")
    except Exception as e:
        print(f"Error creating product: {str(e)}")


def create_multiple_products() -> None:
    """Create multiple products in batch."""
    # Initialize the client with your API key
    client = Worqhat(
        api_key=os.environ.get("WORQHAT_API_KEY"),
        environment=os.environ.get("WORQHAT_ENVIRONMENT", "production"),  # Defaults to production
    )

    try:
        response = client.db.insert_record(
            table="products",       # The table to insert into
            data=[                  # Array of data objects to insert
                {
                    "name": "Basic Widget",
                    "price": 19.99,
                    "inStock": True,
                    "category": "essentials"
                },
                {
                    "name": "Standard Widget",
                    "price": 49.99,
                    "inStock": True,
                    "category": "essentials"
                },
                {
                    "name": "Premium Widget",
                    "price": 99.99,
                    "inStock": False,
                    "category": "premium"
                }
            ]
        )

        # Handle the successful response
        print(f"Inserted {len(response.data)} products")
        print(f"Created products: {response.data}")
    except Exception as e:
        print(f"Error creating products: {str(e)}")


def db_insert() -> None:
    """Run all insert examples."""
    create_user()
    create_product_with_custom_id()
    create_multiple_products()

File: node-typescript/src/endpoints/dbInsert.ts
import Worqhat from "worqhat";

export async function createTask() {
  // Initialize the client with your API key - matching smoke test
  const apiKey = process.env.WORQHAT_API_KEY;
  if (!apiKey) {
    throw new Error("WORQHAT_API_KEY environment variable is required");
  }

  const client = new Worqhat({
    apiKey, // Always use environment variables for API keys
  });

  try {
    const taskId = Math.floor(Date.now() / 1000);
    const nowIso = new Date().toISOString();

    // Call the insertRecord method - matching smoke test
    const response = await client.db.insertRecord({
      table: "tasks", // The table to insert into - matching smoke test
      data: {
        // The data to insert - matching smoke test structure
        id: taskId,
        status: "open",
        priority: "high",
        created_at: nowIso,
        updated_at: nowIso,
      },
    });

    // Handle the successful response
    console.log("Task created with ID:", taskId);
    console.log("Created task:", response.data);
    return response;
  } catch (error) {
    // Handle any errors
    const errorMessage = error instanceof Error ? error.message : String(error);
    console.error("Error creating task:", errorMessage);
    throw error;
  }
}

export async function createProductWithCustomId() {
  // Initialize the client with your API key
  const apiKey = process.env.WORQHAT_API_KEY;
  if (!apiKey) {
    throw new Error("WORQHAT_API_KEY environment variable is required");
  }

  const client = new Worqhat({
    apiKey,
  });

  try {
    // Generate a custom ID
    const customId = `prod_${Date.now()}`;

    // Call the insertRecord method with custom documentId
    const response = await client.db.insertRecord({
      table: "products", // The table to insert into
      data: {
        // The data to insert
        documentId: customId, // Specify your own document ID
        name: "Premium Widget",
        price: 99.99,
        inStock: true,
        category: "electronics",
      },
    });

    // Handle the successful response
    console.log(
      `Product created with custom ID: ${(response.data as any).documentId}`
    );
    return response;
  } catch (error) {
    const errorMessage = error instanceof Error ? error.message : String(error);
    console.error("Error creating product:", errorMessage);
    throw error;
  }
}

// Removed createMultipleProducts to avoid type conflicts - smoke test only uses single inserts

// Export a function to run all examples (for backward compatibility)
export async function dbInsert() {
  // Call the function - matching smoke test
  await createTask();
}

// Sample Response

// Single user insert response:
//  {
//   "success": true,
//   "data": {
//     "documentId": "doc_12345abcde",
//     "name": "John Doe",
//     "email": "john@example.com",
//     "role": "user",
//     "active": true
//   },
//   "message": "Record inserted successfully"
// }

// Product insert with custom ID response:
//  {
//   "success": true,
//   "data": {
//     "documentId": "prod_1627384950",
//     "name": "Premium Widget",
//     "price": 99.99,
//     "inStock": true,
//     "category": "electronics"
//   },
//   "message": "Record inserted successfully"
// }

// Bulk products insert response:
//  {
//   "success": true,
//   "data": [
//     {
//       "documentId": "doc_12345abcde",
//       "name": "Basic Widget",
//       "price": 19.99,
//       "inStock": true,
//       "category": "essentials"
//     },
//     {
//       "documentId": "doc_67890fghij",
//       "name": "Standard Widget",
//       "price": 49.99,
//       "inStock": true,
//       "category": "essentials"
//     },
//     {
//       "documentId": "doc_54321klmno",
//       "name": "Premium Widget",
//       "price": 99.99,
//       "inStock": false,
//       "category": "premium"
//     }
//   ],
//   "message": "3 records inserted successfully"
// }
